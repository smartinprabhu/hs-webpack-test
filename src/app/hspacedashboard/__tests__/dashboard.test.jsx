import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { useSelector } from 'react-redux';
import MockDate from 'mockdate';
import Dashboard from '../dashboard';
import 'babel-polyfill';

jest.mock('react-redux', () => ({
  useSelector: jest.fn((fn) => fn()),
  useDispatch: () => jest.fn(),
}));
jest.mock('@fullcalendar/react', () => { });
jest.mock('@fullcalendar/timegrid', () => { });
jest.mock('@fullcalendar/interaction', () => { });
jest.mock('../../booking/createBookingViews/floorView/layoutFloorView', () => { });
jest.mock('../../booking/createBooking/workStationTypes', () => { });
jest.mock('../../analytics/analyticsView', () => { });

afterEach(cleanup);

// beforeAll(() => {
//   process.env = Object.assign(process.env, { ONESIGNALAPPID: '1234' });
// });

const userInfo = {
  data: {
    allowed_companies: [{
      country: { id: 104, name: 'India' },
      id: 3,
      locality: 'Bengaluru',
      name: 'Divyasree Technopark..',
      postal_code: '560037',
      region: { id: 593, name: 'Karnataka (IN)' },
      street_address: 'DTP Entry Gate, EPIP Zone',
    }],
    employee: {
      id: 6235,
      is_onboarded: false,
      name: 'Raja',
      registration_status: 'draft',
    },
    company: {
      address: 'Water Mark  Building\n222 W. Las Colinas Blvd.\nSuite 755 East\nIrving TX 75039\nUnited States',
      id: 3,
      name: 'Water Mark  Building',
      timezone: 'US/Central',
    },
    id: 1114,
    image_url: 'https://demo.helixsense.com/web/image/res.partner/1455/image/1455.png',
    locale: 'es_ES',
    main_company: { id: 63, name: 'Water Mark Building' },
    mobile: '',
    name: 'Raja',
    phone_number: '',
    timezone: 'Asia/Kolkata',
    user_role: 'tenant_hr',
    username: 'rajahr@gmail.com',
  },
};

describe('Dashboard testing', () => {
  it('snapshot testing ', async () => {
    MockDate.set(1434319925275);
    useSelector.mockImplementation((fn) => fn({
      user: {
        userInfo: userInfo.data,
      },
      config: {
        covidResources: {
          loading: false,
          err: null,
          data: {},
        },
      },
      configObj:
      {
        loading: false,
        err: null,
        data: {},
      },
      bookingInfo: {
        bookingList: {
          loading: false,
          err: null,
          data: {},
        },
      },
    }));
    const { asFragment } = render(<Dashboard />);
    expect(asFragment()).toMatchSnapshot();
  });
});
