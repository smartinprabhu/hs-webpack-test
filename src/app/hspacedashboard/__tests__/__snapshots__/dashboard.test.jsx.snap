// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard testing snapshot testing  1`] = `
<DocumentFragment>
  <div
    class="h-100 row"
  >
    <div
      class="mt-2 col-12 col-sm-7 col-md-7 col-lg-7 col-xl-7"
    >
      <div
        class="bg-azure h-100 rounded p-3 booking dashboardTileHeight"
      >
        <div>
          <div
            class="mx-0 row"
          >
            <div
              class="pl-0 col-3 col-sm-2 col-md-2 col-lg-2"
            >
              <span
                class="image-border"
              >
                <img
                  alt="user"
                  class="border-radius-50 dashboard-user-image user-circle"
                  data-testid="user"
                  src="https://demo.helixsense.com/web/image/res.partner/1455/image/1455.png"
                />
              </span>
            </div>
            <div
              class="pl-0 col-9 col-sm-10 col-md-10 col-lg-10"
            >
              <h1
                class="mt-2"
              >
                 Hello
                <span
                  class="ml-2"
                >
                  Raja!
                </span>
              </h1>
            </div>
          </div>
          <div
            class="mx-0 mt-3 row"
          >
            <div
              class="h-275 pl-0 pr-2 col-12 col-sm-8 col-md-8 col-lg-8"
            >
              <div
                class="rounded noBorder h-100 card"
              >
                <div
                  class="p-2 card-body"
                >
                  <div
                    class="mr-0 row"
                  >
                    <div
                      class="text-center col-2 col-sm-2 col-md-2 col-lg-2 col-xl-2"
                    >
                      <div
                        class="bg-white border-azure-1px border-radius75 pt-2 br-8px card-height  card"
                      >
                        <small
                          class="text-light-cloud-burst font-size-xsmall"
                        >
                          Sunday
                        </small>
                        <span
                          class="text-mandy font-size-16 mb-0"
                        >
                          14
                        </span>
                      </div>
                    </div>
                    <div
                      class="px-1 col-10 col-sm-10 col-md-10 col-lg-10 col-xl-10"
                    >
                      <div
                        class="mt-2"
                      >
                        <span
                          class="h3 font-weight-700"
                        >
                          MY BOOKINGS
                        </span>
                      </div>
                      <hr
                        class="m-0"
                      />
                    </div>
                  </div>
                  <div
                    class="text-center"
                  >
                    <div
                      class="text-center mt-6 font-weight-300"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="bg-white rounded p-2 pt-3 h-275 col-12 col-sm-4 col-md-4 col-lg-4"
            >
              <h4>
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-phone-alt fa-w-16 "
                  data-icon="phone-alt"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M497.39 361.8l-112-48a24 24 0 0 0-28 6.9l-49.6 60.6A370.66 370.66 0 0 1 130.6 204.11l60.6-49.6a23.94 23.94 0 0 0 6.9-28l-48-112A24.16 24.16 0 0 0 122.6.61l-104 24A24 24 0 0 0 0 48c0 256.5 207.9 464 464 464a24 24 0 0 0 23.4-18.6l24-104a24.29 24.29 0 0 0-14.01-27.6z"
                    fill="currentColor"
                  />
                </svg>
                <span
                  class="ml-2 font-weight-800"
                >
                  IMPORTANT CONTACTS
                </span>
              </h4>
              <div
                class="pl-2 mt-2"
              >
                <div
                  class="font-weight-700"
                >
                  IT Service Desk
                </div>
                <div
                  class="font-weight-400"
                />
                <div
                  class="font-weight-400"
                />
                <div
                  class="light-text"
                >
                  8100 4000 / 8101 4000
                </div>
                <div
                  class="font-weight-400"
                >
                  US: 
                  <span
                    class="light-text"
                  >
                    888-532-6021
                  </span>
                </div>
                <div
                  class="font-weight-400 font-11"
                >
                  Additional toll free numbers: 
                  <a
                    class="font-weight-400"
                    href="https://nttdatagroup.sharepoint.com/sites/SRVS-myHubInfo/SitePages/IT-Support.aspx"
                    rel="noreferrer"
                    target="_blank"
                  >
                    available here
                  </a>
                </div>
                <a
                  class="font-weight-400 font-11"
                  href="https://services-onentt.okta.com/home/nttdataservices_remedycorporateitproduction_1/0oa8j40sdhtWNZnqO2p7/aln8j49l1dZjNHLYm2p7"
                  rel="noreferrer"
                  target="_blank"
                >
                  Submit Service Request
                </a>
                <hr
                  class="my-2"
                />
              </div>
              <div
                class="pl-2 mt-2"
              >
                <div
                  class="font-weight-700"
                >
                  OKC Facilities
                </div>
                <div
                  class="font-weight-400"
                >
                  Bobby Duncan
                </div>
                <div
                  class="font-weight-400"
                >
                  bobby.duncan@nttdata.com
                </div>
                <div
                  class="light-text"
                >
                  +1 588-975-2740
                </div>
                <hr
                  class="my-2"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mt-2 pr-0 col-12 col-sm-5 col-md-5 col-lg-5 col-xl-5"
    >
      <div
        class="bg-azure rounded m-0 p-3 dashboardTileHeight row"
      >
        <div
          class="col-sm-10 col-md-10 col-lg-10"
        >
          <h2>
            My Actions
          </h2>
          <div
            class="mt-3 card"
          >
            <div
              class="p-3 card-body"
            >
              <div
                class="row"
              >
                <div
                  class="col-3 col-sm-2 col-md-2 col-lg-2"
                >
                  <img
                    alt="actions"
                    height="45"
                    src="[object Object]"
                    width="45"
                  />
                </div>
                <div
                  class="pl-0 col-7 col-sm-8 col-md-8 col-lg-8"
                >
                  <h3
                    class="mt-2 ml-2 cursor-pointer font-weight-700 pr-0"
                  >
                    Bookings in calendar view
                  </h3>
                </div>
                <div
                  class="col-2 col-sm-2 col-md-2 col-lg-2"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root float-right mt-2 font-size-30px"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="mt-10 ml-11 mb-2 p-0 float-right col-sm-12"
        >
          <img
            alt="Helixsense Portal"
            class="mt-auto w-inherit"
            height="140"
            src="[object Object]"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
